components/generic_modal.tsx

```diff
@@ -130,7 +131,8 @@ export default class GenericModal extends React.PureComponent<Props, State> {
                 enforceFocus={this.props.enforceFocus}
                 restoreFocus={true}
                 role='dialog'
-                aria-labelledby='genericModalLabel'
+                aria-label={this.props.ariaLabel}
+                aria-labelledby={this.props.ariaLabel ? undefined : 'genericModalLabel'}
```

Comment:
Could do `this.props.ariaLabel || 'genericModalLabel'`, 0/5

Reply:
I would've preferred do to something like that as well, but it guarantees that `aria-labelledby` is always defined which would take priority over `aria-label`. The issue stems from `aria-labelledby` not being able to read the value from the `FormattedMessage` component that is used for the header, so I just give priority to `aria-label` for components that use it to resolve the issue.